src/main/python/main.py

```diff
@@ -0,0 +1,1123 @@
+from de.fraunhofer.aisec.cpg.graph.statements.expressions import CallExpression
+from de.fraunhofer.aisec.cpg.graph.statements.expressions import ConstructExpression
+from de.fraunhofer.aisec.cpg.graph.statements.expressions import DeclaredReferenceExpression
+from de.fraunhofer.aisec.cpg.graph.statements.expressions import Expression
+from de.fraunhofer.aisec.cpg.graph.statements.expressions import Literal
+from de.fraunhofer.aisec.cpg.graph.statements.expressions import MemberCallExpression
+from de.fraunhofer.aisec.cpg.graph.statements.expressions import MemberExpression
+from de.fraunhofer.aisec.cpg.graph.statements.expressions import UnaryOperator
+from de.fraunhofer.aisec.cpg.graph.statements.expressions import InitializerListExpression
+from de.fraunhofer.aisec.cpg.graph.types import TypeParser
+from de.fraunhofer.aisec.cpg.sarif import PhysicalLocation
+from de.fraunhofer.aisec.cpg.sarif import Region
+from java.net import URI
+from java.util import ArrayList
+
+
+#############################
+# PROBLEMS / OPEN QUESTIONS #
+#############################
+# 1. Wie gehen wir mit Expression außerhalb von Funktionen um? Für Funktionen gibt es die FunctionDeclaration. Was gibt es für Expressions?
+# 2. File Info kaputt
+# 3. import == include???
+# 4. Import from??? Import alias???
+# 5. default args: wip für cpp
+# 6. Exceptions -> unary op
+# 7. listen / tupel?
+# 8. 2 edges func -> pvd (scopemanager addDeclaration zu oft?)
+# 9. self -> go receiver vom christian angucken
+# 10. assign -> neue decl mit initializer (statt in visit_Name)
+# 11. test Java isinstance aktuell mit java_name startswith -> :( -> chrisitan
+# 12. function.py -> visit_return 2* ???
+
+def debug_print(string, level=1):
+    callerframerecord = inspect.stack()[level]
+    frame = callerframerecord[0]
+    info = inspect.getframeinfo(frame)
+    print("%s\t%d:\t%s" % (info.function, info.lineno, string))
+
+
+class CodeExtractor:
+    # Simple/ugly class to extrace code snippets given a region
+    def __init__(self, fname):
+        with open(fname) as f:
+            self.lines = f.read().splitlines()
+
+    def get_snippet(self, lineno, col_offset, end_lineno, end_col_offset):
+        # 1 vs 0-based indexing
+        lineno -= 1
+        # col_offset -= 1
+        end_lineno -= 1
+        # end_col_offset -= 1
+        if lineno == end_lineno:
+            return self.lines[lineno][col_offset:end_col_offset]
+        else:
+            res = []
+            # first line is partially read
+            res.append(self.lines[lineno][col_offset:])
+            lineno += 1
+
+            # fill with compelte lines
+            while lineno + 1 < end_lineno:
+                res.append(self.lines[lineno][:])
+                lineno += 1
+
+            # last line is partially read
+            res.append(self.lines[end_lineno][:end_col_offset])
+
+            return "\n".join(res)
+
+
+class MyWalker(ast.NodeVisitor):
```

Comment:
Rename MyWalker to something more meaningful

Reply:
In the long run, we might want to split it into separate files and walkers. Let's leave that for a future PR.